### PR TITLE
Upgrade to React 16

### PR DESCRIPTION
**Pull request by entrptaher**
This commit will provide drop in replacements for the migration purpose of the following packages.
- react
- react-dom
- react-komposer

It will use following replacements:
- prop-types
- create-react-class
- composeWithTracker

As well as suppress several errors regarding inline editing and database access.

### CODE REVIEW

File: imports/ui/pages/flowdesigner.js [hunk 1]
```diff
@@ -261,18 +265,18 @@ export class FlowDesigner extends React.Component {
                 </a>
               </li>
               <li className="tab col s2">
-                <a href="http://wireflow.co/mycharts">My charts</a>
+                <a href="/mycharts">My charts</a>
```
Comment (tomscholz) [inline imports/ui/pages/flowdesigner.js]: What about using the [`Link` component](https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md) `react-router` provides?

File: imports/lib/composeWithTracker.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+//
+// compose-with-tracker.js
+//
+import {compose} from "react-komposer";
```
Comment (tomscholz) [inline imports/lib/composeWithTracker.js]: React Komposer hasn't been updated in over a year. Why not use recompose's [`compose`](https://github.com/acdlite/recompose/blob/master/docs/API.md#compose)?

File: imports/ui/components/chat-box.js [hunk 3]
```diff
@@ -108,9 +109,9 @@ class ChatBox extends React.Component {
 }
 
 ChatBox.propTypes = {
-  chats: React.PropTypes.array,
-  routeParams: React.PropTypes.string,
-  wireid: React.PropTypes.string,
+  chats: PropTypes.array,
+  routeParams: PropTypes.string,
+  wireid: PropTypes.string,
```
Comment (tomscholz) [inline imports/ui/components/chat-box.js]: Why not use camelcase here as well? E.g: wireId

Comment (entrptaher) [reply imports/ui/components/chat-box.js]: Check the prop-types package. It's the same reason we write React with capital R. https://www.npmjs.com/package/prop-types

File: imports/lib/composeWithTracker.js (hunk 2 above)
Comment (entrptaher) [reply imports/lib/composeWithTracker.js]: I have just used a drop in replacement to slowly make it compatible with react v16.

File: imports/ui/pages/flowdesigner.js (hunk 1 above)
Comment (entrptaher) [reply imports/ui/pages/flowdesigner.js]: Good thinking. I'll look into that.

**Comment by entrptaher** [comment]
@tomscholz, do you think it'll break or is it acceptable?

File: imports/ui/components/chat-box.js (hunk 3 above)
Comment (entrptaher) [reply imports/ui/components/chat-box.js]: I misunderstood. you meant using `wireId` etc. I'll take a look about that. That's a different issue than PropTypes.

**Comment by tomscholz** [comment]
@entrptaher It looks good to me, for the time being :) The things I *suggested*, where in fact, just suggestions :-)

File: imports/lib/composeWithTracker.js (hunk 2 above)
Comment (badis) [reply imports/lib/composeWithTracker.js]: I think it's better to remove react-komposer entirely and use only withTracker from react-meteor-data.
It's simple and efficient.

**Pull request merged**